patches/minecraft/net/minecraft/item/BlockItem.java.patch

```diff
@@ -9,7 +31,17 @@
                 world.func_184133_a(playerentity, blockpos, this.func_219983_a(blockstate1), SoundCategory.BLOCKS, (soundtype.func_185843_a() + 1.0F) / 2.0F, soundtype.func_185847_b() * 0.8F);
                 itemstack.func_190918_g(1);
                 return ActionResultType.SUCCESS;
-@@ -189,10 +189,18 @@
+@@ -137,6 +140,9 @@
+    }
+ 
+    protected boolean func_195941_b(BlockItemUseContext p_195941_1_, BlockState p_195941_2_) {
++      if (p_195941_1_.func_195991_k().canAddBlockState(p_195941_1_.func_195995_a(), p_195941_2_) && p_195941_1_.func_195991_k().addBlockState(p_195941_1_.func_195995_a(), p_195941_2_, 11)) return true;
++      if (p_195941_1_ instanceof net.minecraftforge.common.util.SameBlockPlaceContext) return false;
++      if (!p_195941_1_.func_196012_c() && !p_195941_1_.func_195991_k().func_180495_p(p_195941_1_.func_195995_a()).func_196953_a(p_195941_1_)) return false;
+       return p_195941_1_.func_195991_k().func_180501_a(p_195941_1_.func_195995_a(), p_195941_2_, 11);
```

Comment:
I'd like feedback on this method.
Implementation-wise it's correct, but I have my doubts about modifying a vanilla method to add extra checks (especially considering they're relocated from `tryPlace`) and feel like it might be better to create an entirely new method instead.
Thoughts?